### PR TITLE
chore: curve fetcher review

### PR DESCRIPTION
**Pull request by koonopek**

### CODE REVIEW

File: packages/oracle-node/src/fetchers/curve/CurveFetcher.ts [hunk 1]
```diff
@@ -33,41 +29,40 @@ export class CurveFetcher extends DexOnChainFetcher<Response> {
     super(name);
   }
 
-  async makeRequest(id: string): Promise<Response> {
-    try {
```
Comment (koonopek) [inline packages/oracle-node/src/fetchers/curve/CurveFetcher.ts:37]: this try/catch is redundant here, this case will be handled by `MultiRequestFetcher)

File: packages/oracle-node/src/fetchers/curve/CurveFetcher.ts [hunk 2]
```diff
@@ -33,41 +29,40 @@ export class CurveFetcher extends DexOnChainFetcher<Response> {
...
-    return response.assetId;
-  }
-
-  validateResponse(response: Responses<Response>): boolean {
-    return response !== undefined;
+    return {
+      ratio: ratio.div(CURVE_PRECISION_RATIO.div(ratioMultiplier)),
+      assetId,
+    };
   }
```
Comment (koonopek) [inline packages/oracle-node/src/fetchers/curve/CurveFetcher.ts:61]: Comment for `calculateLiquidity`:
are we sure we want to return 0 here
it seems very weird, maybe we should just throw an error or override `DexOnChainFetcher::extractPrice`
or even return undefined
throwing error is the best solution in my opinion

File: packages/oracle-node/src/fetchers/MultiRequestFetcher.ts [hunk 3]
```diff
@@ -50,7 +50,6 @@ export abstract class MultiRequestFetcher extends BaseFetcher {
     for (const requestId of requestIds) {
       promises.push(this.makeSafeRequest(requestId));
     }
-
```
Comment (Lukasz2891) [inline packages/oracle-node/src/fetchers/MultiRequestFetcher.ts:53]: why did you remove the empty line before `return` ;p?

File: packages/oracle-node/src/fetchers/curve/CurveFetcher.ts [hunk 4]
```diff
@@ -33,42 +29,42 @@ export class CurveFetcher extends DexOnChainFetcher<Response> {
...
+    const curveFactory = new Contract(address, abi, provider);
 
-      const ratio = await curveFactory[functionName](
-        tokenIndex,
-        pairedTokenIndex,
-        (DEFAULT_RATIO_QUANTITY * ratioMultiplier).toString()
-      );
+    const ratio = (await curveFactory[functionName](
+      tokenIndex,
+      pairedTokenIndex,
```
Comment (koonopek) [inline packages/oracle-node/src/fetchers/curve/CurveFetcher.ts:46]: Please check this out this is the biggest change here, very important

File: packages/oracle-node/src/fetchers/curve/CurveFetcher.ts [hunk 5]
```diff
@@ -33,42 +29,42 @@ export class CurveFetcher extends DexOnChainFetcher<Response> {
...
-      return {
-        ratio,
-        assetId: id,
-      };
-    } catch (error) {
-      throw error;
-    }
-  }
-
-  getAssetId(response: Response) {
```
Comment (koonopek) [inline packages/oracle-node/src/fetchers/curve/CurveFetcher.ts:59]: It is not used anywhere

File: packages/oracle-node/src/fetchers/curve/CurveFetcher.ts [hunk 6]
```diff
@@ -33,42 +29,42 @@ export class CurveFetcher extends DexOnChainFetcher<Response> {
...
-    return response.assetId;
-  }
-
-  validateResponse(response: Responses<Response>): boolean {
-    return response !== undefined;
+    return {
+      ratio:
+        Number(ratio.toString()) / (CURVE_PRECISION_RATIO / ratioMultiplier),
```
[the remark this answers is not in the document]
Comment (koonopek) [reply packages/oracle-node/src/fetchers/curve/CurveFetcher.ts]: Done

File: packages/oracle-node/src/fetchers/curve/CurveFetcher.ts (hunk 5 above)
Comment (koonopek) [reply packages/oracle-node/src/fetchers/curve/CurveFetcher.ts]: removed

File: packages/oracle-node/src/fetchers/curve/CurveFetcher.ts (hunk 2 above)
Comment (koonopek) [reply packages/oracle-node/src/fetchers/curve/CurveFetcher.ts:57]: done

File: packages/oracle-node/src/fetchers/curve/CurveFetcher.ts (hunk 1 above)
Comment (koonopek) [reply packages/oracle-node/src/fetchers/curve/CurveFetcher.ts]: removed

File: packages/oracle-node/src/fetchers/curve/CurveFetcher.ts [hunk 7]
```diff
@@ -20,52 +26,59 @@ export class CurveFetcher extends DexOnChainFetcher<CurveFetcherResponse> {
...
       { blockTag }
-    );
+    )) as BigNumber;
 
     return {
-      ratio,
-      assetId: id,
+      ratio: new Decimal(ratio.toString()).div(
+        CURVE_PRECISION_DECIMAL.div(ratioMultiplier)
+      ),
```
Comment (cehali) [inline packages/oracle-node/src/fetchers/curve/CurveFetcher.ts:53]: I am thinking just to be easier to understand I would extract it to separate variable and maybe split operations. Something like:
`js
const ratioWithDefaultPrecision = new Decimal(ratio.toString()).div(
        CURVE_PRECISION_DECIMAL)
const ratioWithAdditionalMultiplier = ratioWithDefaultPrecision.div(ratioMultiplier);
`
or
`js
const precisionDivier = CURVE_PRECISION_DECIMAL.mul(ratioMultiplier)
const ratioWithPrecision = new Decimal(ratio.toString()).div(precisionDivier)
`

File: packages/oracle-node/src/fetchers/curve/CurveFetcher.ts [hunk 8]
```diff
@@ -1,14 +1,20 @@
...
+import { PoolsConfig } from "./curve-fetchers-config";
 
-const DEFAULT_DECIMALS = 8;
-const DEFAULT_RATIO_QUANTITY = 10 ** DEFAULT_DECIMALS;
+// as default value we should use 10 ** 18 this the PRECISION on Curve StableSwap contract
+// if we provide value lower then 10 ** 18 we start loosing PRECISION
+// if we provide value bigger then 10 ** 18  for example 100 * 10**18 we are saying how much 100 x  pairedTokenIndex i wil receive for tokenIndex
+// thus there is higher chance that slippage will occur and will affect price
+// same in examples https://curve.readthedocs.io/factory-pools.html#StableSwap.get_dy and on Curve frontend
+// in case of LPs (with big volume) it shouldn't be a problem, however this just not correct
```
Comment (cehali) [inline packages/oracle-node/src/fetchers/curve/CurveFetcher.ts:13]: WDYT to use `/* */` for long comments?

**Review by cehali — DISMISSED** [review]
Good job, two minor comments

File: packages/oracle-node/src/fetchers/curve/CurveFetcher.ts (hunk 7 above)
Comment (koonopek) [reply packages/oracle-node/src/fetchers/curve/CurveFetcher.ts:53]: Good point!

**Review by hatskier — APPROVED** [review]
